From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 27 Nov 2017 16:24:53 +0000
Subject: dm: Avoid ABI change in 3.16.50
Forwarded: not-needed

Commit 604407890ecf "dm: fix printk() rate limiting code" removed the
global state for device-mapper log rate limiting.  Add it back so any
old OOT modules using it will still work.

---
--- a/drivers/md/dm.c
+++ b/drivers/md/dm.c
@@ -24,6 +24,16 @@
 
 #define DM_MSG_PREFIX "core"
 
+#ifdef CONFIG_PRINTK
+/*
+ * ratelimit state to be used in DMXXX_LIMIT() (old modules only)
+ */
+DEFINE_RATELIMIT_STATE(dm_ratelimit_state,
+		       DEFAULT_RATELIMIT_INTERVAL,
+		       DEFAULT_RATELIMIT_BURST);
+EXPORT_SYMBOL(dm_ratelimit_state);
+#endif
+
 /*
  * Cookies are numeric values sent with CHANGE and REMOVE
  * uevents while resuming, removing or renaming the device.
